src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -564,6 +595,27 @@ private static bool ParsePackageAndGenerateInstallerNodes(InstallerMetadata inst
             }
 
             return archMatches.Count == 1 ? archMatches.Single() : null;
+        }
+
+        /// <summary>
+        /// Performs a regex match to determine the installer scope based on the url string.
+        /// </summary>
+        /// <param name="url">Installer url string.</param>
+        /// <returns>Installer scope enum.</returns>
+        private static Scope? GetScopeFromUrl(string url)
+        {
+            if (Regex.Match(url, "user", RegexOptions.IgnoreCase).Success)
+            {
+                return Scope.User;
+            }
+            else if (Regex.Match(url, "machine", RegexOptions.IgnoreCase).Success)

```

Comment:
Added 'allusers` to the `machine` scope condition and reordered the blocks to avoid conflicts. Let me know if there are any other examples of strings that may indicate the scope as I am not too familiar with what is commonly used in the installer URLS.

Reply:
I think the current conditions are sufficient (as I was typing this, I realized we should probably drop the 's' and change to `alluser` for a more lenient lookup). The scope info is rarely visible in the URL though since majority publishers publish a single installer with InstallerSwitches to alter the scope behavior. In fact, a quick search on all the InstallerUrls in the winget-pkgs repo revealed that the above example is the only one for a machine-wide installer that had scope info in the URL. There are a few others that have `user` in the URL (namely Microsoft.AzureDataStudio ones)

Reply:
After searching through the InstallerUrls though (I ripgrep'ed through them), I found that this logic is likely to cause more conflicts than matches - A couple examples of some conflicting ones are:

-  https:// desktop.github`user`content.com/github-desktop/releases/3.0.2-11fd584c/GitHubDesktopSetup-x64.msi (ID: GitHub.GitHubDesktop)
- https:// raw.github`user`content.com/Pingus/downloads/master/Pingus-0.7.6.exe (ID: IngoRuhnke.Pingus)
- https:// ausweisapp.bund.de/fileadmin/`user`_upload/Software/AusweisApp2-1.24.1.msi (ID: Governikus.AusweisApp2)
- https:// desktop.`user`api.com/win64/master/vk_setup.exe (ID: VKontakte.VKMessenger)
-  https:// github.com/jope`machine`/arvis/releases/download/v0.10.7/Arvis-Setup-0.10.7.exe (ID: jopemachine.Arvis)
- https:// download.no`machine`.com/download/7.7/Windows/nomachine_7.7.4_1.exe (ID: NoMachine.NoMachine)

Some of the correctly matching ones are
- https:// sqlopsbuilds.azureedge.net/stable/2413919f186f780f0193d047da3d90bb3c1e9bf6/azuredatastudio-windows-`user`-setup-1.21.0.exe (ID: Microsoft.AzureDataStudio)
- https:// github.com/namazso/OpenHashTab/releases/download/v3.0.1/OpenHashTab_setup_`user`.exe (ID: namazso.OpenHashTab)
- https:// github.com/joncampbell123/dosbox-x/releases/download/dosbox-x-v0.83.15/dosbox-x-windows-0.83.15-setup-`allusers`.exe (ID: joncampbell123.DOSBox-X)

I think a more precise way is to search for `_user`, `-user` etc. in the URL instead of the raw string itself

Reply:
Thanks for this investigation @mdanish-kh, this is super helpful! I worry now that this may cause more harm than good. I am leaning towards removing this behavior entirely as it does not seem to be as deterministic as what we did for detecting the architecture in the URL...

Reply:
I'd agree as there are considerably more chances of getting a false positive than a deterministic match with the current behavior

Reply:
Removed scope detection from URL